src/table.c

```diff
@@ -116,26 +125,29 @@ static void **jl_table_peek_bp(jl_array_t *a, void *key)
     do {
         if (tab[index] == NULL)
             return NULL;
-        if (jl_egal((jl_value_t*)key, (jl_value_t*)tab[index]))
-            return &tab[index+1];
+        if (jl_egal((jl_value_t *)key, (jl_value_t *)tab[index])) {
+            if (tab[index + 1] != NULL)
+                return &tab[index + 1];
+            if (key != jl_nothing)
+                // `nothing` is our sentinel value for deletion, so need to keep searching if it's also our search key
+                return NULL;
```

Comment:
This looks like dead code --- if `tab[index + 1] == NULL` then `key` must be `nothing`.

Reply:
Good point (For comparison, I put an assert here in the other function. Just didn’t realize I had converged them after exploring some alternative ways of expressing this.)